presto-spi/src/main/java/com/facebook/presto/spi/connector/Connector.java

```diff
@@ -93,6 +93,14 @@ default ConnectorMetadataUpdaterProvider getConnectorMetadataUpdaterProvider()
         throw new UnsupportedOperationException();
     }
 
+    /**
+     * @throws UnsupportedOperationException if this connector does not support metadata update handle serde
+     */
+    default ConnectorMetadataUpdateHandleSerdeProvider getConnectorMetadataUpdateHandleSerdeProvider()
+    {
+        throw new UnsupportedOperationException();
+    }
```

Comment:
Do we expect more handles to be serialized and deserialized beyond `MetadataUpdate`? If that is the case, shall we wrap all handle serdes into one big provider? Because otherwise, we will need to enumerate a lot of serde at the connector level. Check `ConnectorPlanOptimizerProvider` and `ConnectorPlanOptimizer` as an example

Reply:
we will have more handles to support.
`ConnectorSplit` and `ConnectorTransactionHandle` are the ones as a part of `TaskUpdateRequest` conversion.

Reply:
refactored the code to go with a single `ConnectorTypeSerdeProvider` instead of providers for each type.